libs/sdk-backend-tiger/src/backend/workspaces/index.ts

```diff
@@ -57,86 +58,67 @@ class TigerWorkspaceQuery implements IWorkspacesQuery {
         return this;
     }
 
+    public withFilter(filter: {
+        description?: string;
+        earlyAccess?: string;
+        prefix?: string;
+    }): IWorkspacesQuery {
+        this.filter = filter;
+        return this;
+    }
+
     public withSearch(search: string): IWorkspacesQuery {
         this.search = search;
         return this;
     }
 
     public query(): Promise<IWorkspacesQueryResult> {
-        return this.queryWorker(this.offset, this.limit, this.search);
+        return ServerPaging.for(
+            async ({ limit, offset, totalCount }) => {
+                const result = await this.authCall((client) =>
+                    client.entities.getAllEntitiesWorkspaces({
+                        size: limit,
+                        page: offset / limit,
+                        filter: this.constructFilter(),
+                        metaInclude: totalCount === undefined ? ["page"] : undefined,
+                    }),
+                );
+
+                const descriptors = this.resultToWorkspaceDescriptors(result.data);
+                const workspaces = this.descriptorsToAnalyticalWorkspaces(descriptors);
+
+                return {
+                    items: workspaces,
+                    totalCount: result.data.meta?.page?.totalElements ?? totalCount ?? workspaces.length,
```

Comment:
It will be missing on first call, I wanted to avoid having metaInclude=page on each call, as it would pointlessly count all entities when we already have the value.

So first call does not have total count, it needs to be added.
Every subsequent page will have the total count

Reply:
I've opted to use -1 as a sentinel value, it should never appear normally, so if it does we at least will know that something went wrong

Reply:
changed it to a non-null assertion